pfcpiface/parse-far.go

```diff
@@ -52,20 +54,21 @@ type far struct {
 	tunnelPort    uint16
 }
 
-func (f *far) printFAR() {
-	log.Println("------------------ FAR ---------------------")
-	log.Println("FAR ID:", f.farID)
-	log.Println("fseID:", f.fseID)
-	log.Println("fseIDIP:", f.fseidIP)
-	log.Println("dstIntf:", f.dstIntf)
-	log.Println("applyAction:", f.applyAction)
-	log.Println("tunnelType:", f.tunnelType)
-	log.Println("tunnelIP4Src:", f.tunnelIP4Src)
-	log.Println("tunnelIP4Dst:", f.tunnelIP4Dst)
-	log.Println("tunnelTEID:", f.tunnelTEID)
-	log.Println("tunnelPort:", f.tunnelPort)
-	log.Println("sendEndMarker:", f.sendEndMarker)
-	log.Println("--------------------------------------------")
+// Satisfies the fmt.Stringer interface
+func (f far) String() string {
+	var b = strings.Builder{}
+	fmt.Println(&b, "FAR ID:", f.farID)
+	fmt.Println(&b, "fseID:", f.fseID)
+	fmt.Println(&b, "fseIDIP:", f.fseidIP)
+	fmt.Println(&b, "dstIntf:", f.dstIntf)
+	fmt.Println(&b, "applyAction:", f.applyAction)
+	fmt.Println(&b, "tunnelType:", f.tunnelType)
+	fmt.Println(&b, "tunnelIP4Src:", f.tunnelIP4Src)
+	fmt.Println(&b, "tunnelIP4Dst:", f.tunnelIP4Dst)
+	fmt.Println(&b, "tunnelTEID:", f.tunnelTEID)
+	fmt.Println(&b, "tunnelPort:", f.tunnelPort)
+	fmt.Println(&b, "sendEndMarker:", f.sendEndMarker)
```

Comment:
Have to check how this actually looks in a log message. I image logrus might strip `"\n"` characters.

Reply:
```
time="2021-08-06T04:56:10Z" level=info msg="create sessions: 1" func=main.main file="/pfcpiface/main.go:184"
 Src Iface: 2
 tunnelIP4Dst: 0.0.0.0
 tunnelTEID: 0
 srcIP: 0.0.0.0
 dstIP: 16.0.0.1
 srcPort: 0
 dstPort: 0
 proto: 0
 Src Iface Mask: 255
 tunnelIP4Dst Mask: 0.0.0.0
 tunnelTEIDMask Mask: 0
 srcIP Mask: 0.0.0.0
 dstIP Mask: 255.255.255.255
 srcPort Mask: 0
 dstPort Mask: 0
 proto Mask: 0
 pdrID: 0
 fseID 805306368
 fseidIP 0
 ctrID: 0
 farID: 0
 qerID: 0
 needDecap: 0
 allocIPFlag: false
time="2021-08-06T04:56:10Z" level=trace func="main.(*bess).sendMsgToUPF" file="/pfcpiface/bess.go:76"
 Src Iface: 2
 tunnelIP4Dst: 198.19.0.1
 tunnelTEID: 2415919104
 srcIP: 0.0.0.0
 dstIP: 0.0.0.0
 srcPort: 0
 dstPort: 0
 proto: 0
 Src Iface Mask: 255
 tunnelIP4Dst Mask: 255.255.255.255
 tunnelTEIDMask Mask: 4294967295
 srcIP Mask: 0.0.0.0
 dstIP Mask: 0.0.0.0
 srcPort Mask: 0
 dstPort Mask: 0
 proto Mask: 0
 pdrID: 0
 fseID 805306368
 fseidIP 0
 ctrID: 0
 farID: 0
 qerID: 0
 needDecap: 1
 allocIPFlag: false
time="2021-08-06T04:56:10Z" level=trace func="main.(*bess).sendMsgToUPF" file="/pfcpiface/bess.go:76"
 Src Iface: 1
 tunnelIP4Dst: 198.18.0.1
 tunnelTEID: 805306368
 srcIP: 16.0.0.1
 dstIP: 0.0.0.0
 srcPort: 0
 dstPort: 0
 proto: 0
 Src Iface Mask: 255
 tunnelIP4Dst Mask: 255.255.255.255
 tunnelTEIDMask Mask: 4294967295
 srcIP Mask: 255.255.255.255
 dstIP Mask: 0.0.0.0
 srcPort Mask: 0
 dstPort Mask: 0
 proto Mask: 0
 pdrID: 0
 fseID 805306368
 fseidIP 0
 ctrID: 0
 farID: 1
 qerID: 1
 needDecap: 1
 allocIPFlag: false
time="2021-08-06T04:56:10Z" level=trace func="main.(*bess).sendMsgToUPF" file="/pfcpiface/bess.go:76"
 Src Iface: 1
 tunnelIP4Dst: 198.18.0.1
 tunnelTEID: 805306368
 srcIP: 0.0.0.0
 dstIP: 9.9.9.9
 srcPort: 0
 dstPort: 0
 proto: 0
 Src Iface Mask: 255
 tunnelIP4Dst Mask: 255.255.255.255
 tunnelTEIDMask Mask: 4294967295
 srcIP Mask: 0.0.0.0
 dstIP Mask: 255.255.255.255
 srcPort Mask: 0
 dstPort Mask: 0
 proto Mask: 0
 pdrID: 0
 fseID 805306368
 fseidIP 0
 ctrID: 0
 farID: 2
 qerID: 2
 needDecap: 1
 allocIPFlag: false
time="2021-08-06T04:56:10Z" level=trace func="main.(*bess).sendMsgToUPF" file="/pfcpiface/bess.go:76"
 FAR ID: 0
 fseID: 805306368
 fseIDIP: 0
 dstIntf: 0
 applyAction: 2
 tunnelType: 1
 tunnelIP4Src: 3323068417
 tunnelIP4Dst: 184615297
 tunnelTEID: 805306368
 tunnelPort: 2152
 sendEndMarker: false
time="2021-08-06T04:56:10Z" level=trace func="main.(*bess).sendMsgToUPF" file="/pfcpiface/bess.go:87"
 FAR ID: 1
 fseID: 805306368
 fseIDIP: 0
 dstIntf: 1
 applyAction: 2
 tunnelType: 0
 tunnelIP4Src: 0
 tunnelIP4Dst: 0
 tunnelTEID: 0
 tunnelPort: 0
 sendEndMarker: false
time="2021-08-06T04:56:10Z" level=trace func="main.(*bess).sendMsgToUPF" file="/pfcpiface/bess.go:87"
 FAR ID: 2
 fseID: 805306368
 fseIDIP: 0
 dstIntf: 1
 applyAction: 2
 tunnelType: 1
 tunnelIP4Src: 3323133953
 tunnelIP4Dst: 218169799
 tunnelTEID: 2415919104
 tunnelPort: 2152
 sendEndMarker: false
time="2021-08-06T04:56:10Z" level=trace func="main.(*bess).sendMsgToUPF" file="/pfcpiface/bess.go:87"
 QER ID: 0
 fseID: 805306368
 QFI: 9
 fseIDIP: 0.0.0.0
 Uplink Status: 0
 Downling Status: 0
 Uplink MBR: 50000
 Downlink MBR: 50000
 Uplink GBR: 50000
 Downlink GBR: 50000
time="2021-08-06T04:56:10Z" level=trace func="main.(*bess).sendMsgToUPF" file="/pfcpiface/bess.go:98"
 QER ID: 1
 fseID: 805306368
 QFI: 8
 fseIDIP: 0.0.0.0
 Uplink Status: 0
 Downling Status: 0
 Uplink MBR: 50000
 Downlink MBR: 50000
 Uplink GBR: 50000
 Downlink GBR: 50000
time="2021-08-06T04:56:10Z" level=trace func="main.(*bess).sendMsgToUPF" file="/pfcpiface/bess.go:98"
 QER ID: 2
 fseID: 805306368
 QFI: 7
 fseIDIP: 0.0.0.0
 Uplink Status: 0
 Downling Status: 0
 Uplink MBR: 50000
 Downlink MBR: 50000
 Uplink GBR: 50000
 Downlink GBR: 50000
time="2021-08-06T04:56:10Z" level=trace func="main.(*bess).sendMsgToUPF" file="/pfcpiface/bess.go:98"
time="2021-08-06T04:56:10Z" level=info msg="Sessions/s: 215.61543004389713" func="main.(*bess).sim" file="/pfcpiface/bess.go:537"
```

Reply:
Lets make the text accompanying the struct member case consistent. Lets fix the formatting decimal/hex/IP as appropriate and consistent across all 3 - PDR/FAR/QER. Some that I could see

fseID
tunnelTEID
tunnelTEIDMask
tunnelIP4Src
tunnelIP4Dst
tunnelTEID

Reply:
After this LGTM